services/bettersocial/fcmToken/sendMultiDeviceNotification.js

```diff
@@ -28,7 +28,15 @@ const sendMultiDeviceNotification = async (
         token: user?.token
       };
 
-      await messaging().send(payload);
+      try {
+        await messaging().send(payload);
+      } catch (error) {
+        if (error.code === 'messaging/registration-token-not-registered') {
+          console.error(`Token ${user?.token} not found: ${error.message}`);
+        } else {
+          console.error(`Failed to send push notification: ${error.message}`);
+        }
+      }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The changes made here are generally good, as they add error handling to the `messaging().send(payload)` function call. However, it's not a good practice to use `console.error` for logging in production code. It would be better to use a dedicated logging library that can provide more control over log levels, formats, and destinations. Here is an example using the `winston` library:

```diff
-          console.error(`Token ${user?.token} not found: ${error.message}`);
+          logger.error(`Token ${user?.token} not found: ${error.message}`);

-          console.error(`Failed to send push notification: ${error.message}`);
+          logger.error(`Failed to send push notification: ${error.message}`);
```

Remember to import the `logger` at the top of your file:

```javascript
const logger = require('winston');
```

This will allow you to manage your logs more effectively, especially when your application grows in complexity.


<!-- This is an auto-generated comment by OSS CodeRabbit -->